src/sql/platform/dialog/dialogTypes.ts

```diff
@@ -9,17 +9,35 @@ import * as sqlops from 'sqlops';
 import { localize } from 'vs/nls';
 import Event, { Emitter } from 'vs/base/common/event';
 
-export class DialogTab {
+export class ModelViewPane {
+	private _valid: boolean = true;
+	private _validityChangedEmitter = new Emitter<boolean>();
+	public readonly onValidityChanged = this._validityChangedEmitter.event;
+
+	public get valid(): boolean {
+		return this._valid;
+	}
+
+	public notifyValidityChanged(valid: boolean) {
```

Comment:
I added a task to #559 to look at how we can do this better. It would be nice if the `ModelViewPane` had a direct reference to its own `ModelView` object so that it could subscribe to validity changed events directly, but that change is probably worthy of its own PR